src/components/Lookup/index.js

```diff
@@ -68,12 +71,15 @@ class Lookup extends Component {
 
     componentDidUpdate(prevProps) {
         const { options: prevOptions } = prevProps;
-        const { options } = this.props;
+        const { options, preferredSelectedOption } = this.props;
         if (prevOptions !== options) {
```

Comment:
we also need to check whether preferredSelectedOption change to recompute focusedItemIndex again, make it in other if condition since in this case we don't need to recompute normalizedOptions3